src/content/blog/starlight-034.mdx

```diff
@@ -0,0 +1,163 @@
+
+## CSS cascade layers
+
+It’s important that Starlight users can easily customize their sites. Historically, clashes between Starlight’s built-in styles and a user’s CSS made the customization experience less smooth than it could have been:
+
+```css
+/* Starlight built-in styles (simplified) */
+:not(h1, h2, h3, h4, h5, h6) + h2 {
+	margin-top: 1.5em;
+}
+
+/* User styles */
+h2 {
+	margin-top: 1em; /* ❌ Doesn’t apply because `h2` is a lower specificity! */
+}
+```
+
+Starlight v0.34 solves conflicts by moving all built-in styles into a dedicated `starlight` [CSS cascade layer](https://css-tricks.com/css-cascade-layers/). This means that **user styles will always take precedence** over default styles and we can wave goodbye to specificity battles. 👋
+
+```css
+/* Starlight built-in styles (simplified) */
+@layer starlight.content {
+	:not(h1, h2, h3, h4, h5, h6) + h2 {
+		margin-top: 1.5em;
+	}
+}
+
+/* User styles */
+h2 {
+	margin-top: 1em; /* ✅ Applies because it’s in the top layer! */
+}
+```
+
+This change also means you can enjoy the benefits of using `@layer` to organize your own CSS without Starlight’s styles always overriding it.
+
+Learn more about using cascade layers in [Starlight’s “CSS & Styling” guide](https://starlight.astro.build/guides/css-and-tailwind/#cascade-layers).
+
+<Note>
+	**With great power comes great responsibility!** We recommend checking your
+	site’s appearance when upgrading to make sure there are no style regressions
+	caused by this change.
+</Note>
+
+## Improved &lt;head&gt; APIs
+
+In Starlight v0.33, we added a new `head` property to our route data object. This unlocks full control over Starlight’s `<head>` tags in [route middleware](https://starlight.astro.build/guides/route-data/#customizing-route-data), including for plugins, allowing you to add tags and filter default tags more easily.
+
+For example, this middleware uses Railway’s demo [Open Graph image API](https://og.railway.com/) to add `og:image` meta tags to every Starlight page:
+
+```js
+// src/routeData.ts
+import { defineRouteMiddleware } from '@astrojs/starlight/route-data';
+
+export const onRequest = defineRouteMiddleware((context) => {
+	const { entry, head } = context.locals.starlightRoute;
+
+	// Create an Open Graph image URL using the current page’s title.
+	const ogImageUrl = new URL(
+		'https://og.railway.com/api/image?fileType=png&layoutName=simple',
+	);
+	ogImageUrl.searchParams.set('text', entry.data.title);
+
+	// Add a `<meta property="og:image">` tag to the current page’s `<head>`.
+	head.push({
+		tag: 'meta',
+		attrs: { property: 'og:image', content: ogImageUrl.href },
```

Comment:
```suggestion
		attrs: { property: 'og:image', content: ogImageUrl.href },
		content: '',
```

Just noticed that `content` [has a default value](https://github.com/withastro/starlight/blob/main/packages/starlight/schemas/head.ts#L12) in our schema which means that when pushing to the `head` route data, `content` is needed at the moment.

Reply:
Oh, that’s a bit annoying 😀 

Let me see if we can fix that quickly.

Reply:
Made a PR: https://github.com/withastro/starlight/pull/3122

Reply:
Amazing 🙌